### ISSUE TITLE
Groups of New Filters Disappearing / Unable To Move Files

### ISSUE THREAD

**Issue by bit-byte1055** [RealRaven2000/quickFilters#78]
(1) Groups of filters have been randomly disappearing, as though quickFilters is resetting to a previously saved state.  This includes newly created, standalone filters, as well as new filters added to existing filters.

(2) Am also unable to move emails from Inbox to Local Folders, or between local folders, once a local folder has reached a random capacity.  This is with both dragging and using the submenu.  I did not experience this problem, prior to installing quickFolders and quickFilters.  I have no idea if quickFilters is the culprit, but will also post on quickFolders

Pop!_OS 21.04
Thunderbird 91.2.1
quickFilters Pro 5.3.2 (registered)
quickFolders Pro 5.8.3 (registered)

**Comment by RealRaven2000** [comment]
thanks - I will look at the QuickFolders issue later on the dedicated Github. I had some issue with THunderbird 68 finding subfolders located in Local Folders, but could not reproduce it on Thunderbird 91. Maybe I can reproduce it on 68 again and find an issue that also sometimes applies to Tb91.

> (1) Groups of filters have been randomly disappearing, as though quickFilters is resetting to a previously saved state. This includes newly created, standalone filters, as well as new filters added to existing filters.

that shouldn't happen - Thunderbird is in charge of storing filters, once you close the message filters window (do you see it after creating a new filter?) it should be stored. Which you can check by re-opening the message filters window manually. If it's still there it should also already be stored by Thunderbird and reappear when restarting Tb. Otherwise it might be a Thunderbird bug.

If you could add a screenshot with an example for a filter you created (showing in the filters list) and then another how it is gone after restart, it may be helpful. Be sure to check that it is stored in the correct account.

**Comment by bit-byte1055** [comment]
Thanks for the quick reply!  I'm not sure, but the file moving problem might have been solved with the latest QuickFolders 5.8.3 update, which I performed yesterday.  I haven't yet posted anything about the problem to GitHub; but will do so, if the problem continues.

Attached is a screen shot of the Message Filters window for my primary email address.  The "Junk ..." filters contain nested filters that move designated emails to the Local Junk Folder.  I've been limiting the "Junk ..." filters to 50 nested filters.

Recently, there were three Junk filters: "Junk 1-50", "Junk 51-100", and "Junk 101-105".  However, "Junk 101-105" disappeared, "Junk 51-100" lost eight of the nested filters, and "Junk 1-50" lost three filters.  I have since repopulated the filters and renamed them, as indicated in the screenshot.

This has been an ongoing, random problem with the filters.  I'm wondering if it might have something to do with: (1) The Repair Folder function, or (2) the number of filters nested within a "parent" filter.

While troubleshooting the file moving problem, I used Repair Folder numerous times.  I didn't notice any correlation between Repair Folder and the disappearing filters, but it might be a possibility, nonetheless.

I'm also wondering if there is a practical limit to the number of nested filters within a "parent" filter (ie "Junk 1-50").

For what it's worth, I'm not a programmer, but have been involved with computers, since the mid-1980's - assembling my own computers and working with DOS, Windows, UNIX, Solaris, Linux, & macOS, including using the command line in all the platforms.

![Screenshot from 2021-11-04 14-11-45](https://user-images.githubusercontent.com/93684121/140413661-2ec234

**Comment by RealRaven2000** [comment]
Could you export your filters for me using the backup button? I am not sure what you mean by "nested" filters - but I am aware that many users create filters with many search terms for combatting spam:

![image](https://user-images.githubusercontent.com/6314674/140493271-f0d0520d-a2d3-4cb9-9eba-0a537cb385bc.png)

(if you are worried about privacy, just email them to me directly, with a link to issu #78 for context)

Tip: It's always a good idea to have less than 100 search term in a filter. I think there is a hard but undocumented limit within Thunderbird .

**Comment by bit-byte1055** [comment]
I'll export & email the filters, later today.  Yeah, my use of "nested" filters is my own semantics for what you're calling "search term".  When creating a filter, one can create a new one or merge it with an existing filter.  So, merged filters actually contain multiple "filters" ("nested" filters).  Perhaps the term "rules" within a filter would be better.

**Comment by RealRaven2000** [comment]
> I'll export & email the filters, later today. Yeah, my use of "nested" filters is my own semantics for what you're calling "search term". When creating a filter, one can create a new one or merge it with an existing filter. So, merged filters actually contain multiple "filters" ("nested" filters). Perhaps the term "rules" within a filter would be better.

thank you. Yes "nested" filters would actually be a filter that relies on the results of another filter, which would be a great technicak solution for the limitation of the Thunderbird UI which doesn't allow  combining the "any search term" and "all search terms" in more complex ways such as  Term A and (Term B or Term C). When you merge filters using quickFilters, you simply expand them with more search terms (i.e. add new search conditions)

**Comment by bit-byte1055** [comment]
Yeah, it's all semantics.  My ISP mail server has "dependent" filters that are as you describe.  Can get quite complex, but are especially nice for sorting-out spoof & other nefarious emails.

**Comment by bit-byte1055** [comment]
Hi, Re: Issue #78 <https://github.com/RealRaven2000/quickFilters/issues/78>

Attached is a backup of the filters for my primary email address.  I 
don't know if it's related; but, after updating QuickFolders on 
11/4/2021, I haven't experienced the problem with missing filters in 
quickFilters.  (I also haven't been experiencing the problem with moving 
files between folders.)

Thanks,
Scott

*******

**Comment by bit-byte1055** [comment]
The problem, of the quickFilters database resetting to a previously saved state, has returned.  What was filter "Junk 101-112" (containing twelve filters), has reverted back to "Junk 101-102" (as it was saved a week or so ago).  When adding a search term to a Junk filter, I update the name, to reflect the number of search terms the filter contains.  So, over the period of a week, "Junk 101-102" was updated ten times, to become "Junk 101-112".  The mechanism that reverts the filter back to "Junk 101-102" is a problem with either quickFilters, Thunderbird, or the OS (Pop!_OS 21.04).

Pop!_OS 21.04
Thunderbird 91.3.0
quickFilters Pro 5.3.2 (registered)
quickFolders Pro 5.8.3 (registered)

**Comment by RealRaven2000** [comment]
Pretty sure it is not quickFIlters (it doesn't save or load unless you do a backup)
Pretty sure it isn't Thunderbird as it hasn't got a "time machine" function that restores old files - conversion and changes always go forward. If a change is saved once to hard drive there is no way to get back.

So that leaves your operating system / antivirus / external backup systems as possible explanations.
